airflow/api_fastapi/core_api/routes/public/providers.py

```diff
@@ -0,0 +1,55 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import re2
+
+from airflow.api_fastapi.common.parameters import QueryLimit, QueryOffset
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.serializers.providers import ProviderCollectionResponse, ProviderResponse
+from airflow.providers_manager import ProviderInfo, ProvidersManager
+
+providers_router = AirflowRouter(tags=["Provider"], prefix="/providers")
+
+
+def _remove_rst_syntax(value: str) -> str:
+    return re2.sub("[`_<>]", "", value.strip(" \n."))
+
+
+def _provider_mapper(provider: ProviderInfo) -> ProviderResponse:
+    return ProviderResponse(
+        package_name=provider.data["package-name"],
```

Comment:
Do we want to expose anything else in `provider.data` as  a dict?

Reply:
Yes, maybe not in the first shot. But we decided to NOT use the previous FAB connection forms... and in this regard it would make sense to return the providers hooks (in this service or separate?) as connection extra form fields... such that w/o FAB dependency the connection forms can be rendered... but maybe/mostly this should be extended later... not putting this on-hold.

Anyway - other meta data to expose... like plugin list, operators, hooks... etc what is contained in provider?

Reply:
Yes we can definitely add more depending on what we need.

Let me know what you would like to see here and I can add it in a follow up PR. It looks like there are a lot of different types of `data` and that might be the reason why a plain 'dict' was specified, without trying to narrow the type.